File: customer-java-resttemplate/src/main/java/com/prez/api/dto/CustomerPreferencesResponse.java
package com.prez.api.dto;

import static java.util.stream.Collectors.toList;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.prez.model.CustomerPreferences;
import java.util.List;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class CustomerPreferencesResponse {
  List<CustomerPreferencesProfileResponse> profiles;

  public static CustomerPreferencesResponse of(List<CustomerPreferences> customerPreferences) {
    List<CustomerPreferencesProfileResponse> profiles = customerPreferences.stream()
        .map(profiled -> CustomerPreferencesProfileResponse.builder()
            .id(profiled.getId())
            .customerId(profiled.getCustomerId())
            .seatPreference(profiled.getSeatPreference())
            .classPreference(profiled.getClassPreference())
            .profileName(profiled.getProfileName())
            .language(profiled.getLanguage())
            .build()).collect(toList());
    return CustomerPreferencesResponse.builder().profiles(profiles).build();
  }
}

File: customer-kotlin-coroutines-controllers/src/main/kotlin/com/prez/api/dto/CustomerPreferencesResponse.kt
package com.prez.api.dto

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonInclude
import com.prez.model.SeatPreference
import java.util.*

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class CustomerPreferencesResponse (
  val profiles: List<CustomerPreferencesProfileResponse>
)

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class CustomerPreferencesProfileResponse(
  val id: String,
  val customerId: String,
  @JsonFormat(shape = JsonFormat.Shape.STRING) val seatPreference: SeatPreference,
  val classPreference: Int,
  val profileName: String,
  val language: Locale?
)